scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -569,13 +571,35 @@ public void sessionEstablished(Handshaker handshaker, DTLSSession session) throw
 
 		@Override
 		public void handshakeCompleted(Handshaker handshaker) {
+			SerialExecutor executor = serialExecutor;
+			if (executor != null && !executor.isShutdown()) {
+				try {
+					executor.assertOwner();
+				} catch (ConcurrentModificationException ex) {
+					LOGGER_OWNER.error("on handshake completed: connection {}", ex.getMessage(), ex);
+					if (LOGGER_OWNER.isDebugEnabled()) {
+						throw ex;
+					}
+				}
+			}
 			if (ongoingHandshake.compareAndSet(handshaker, null)) {
 				LOGGER.debug("Handshake with [{}] has been completed", handshaker.getPeerAddress());
 			}
 		}
 
 		@Override
 		public void handshakeFailed(Handshaker handshaker, Throwable error) {
+			SerialExecutor executor = serialExecutor;
+			if (executor != null && !executor.isShutdown()) {
```

Comment:
should we make the check only available is LOGGER_OWNER is ErrorEnabled ? (same for handshake complete)

Reply:
I can add the check, but I would recommend to have it enabled.